src/main/java/ch/njol/skript/variables/JdbcStorage.java

```diff
@@ -0,0 +1,553 @@
+package ch.njol.skript.variables;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import org.jetbrains.annotations.Nullable;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptAddon;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.log.SkriptLogger;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.skript.util.Task;
+import ch.njol.skript.util.Timespan;
+import ch.njol.skript.util.Timespan.TimePeriod;
+import ch.njol.util.SynchronizedReference;
+
+/**
+ * A storage for Skript variables that uses a SQL database.
+ * <p>
+ * This class is abstract and should be extended to implement specific SQL database storage.
+ * 
+ * The queries will select and delete variables into the database.
+ */
+public abstract class JdbcStorage extends VariableStorage {
+
+	protected static final String DEFAULT_TABLE_NAME = "variables21";
+
+	public static final int MAX_VARIABLE_NAME_LENGTH = 380; // MySQL: 767 bytes max; cannot set max bytes, only max characters
+	public static final int MAX_CLASS_CODENAME_LENGTH = 50; // checked when registering a class
+	public static final int MAX_VALUE_SIZE = 10000;
+
+	/**
+	 * Params: name, type, value
+	 * <p>
+	 * Writes a variable to the database
+	 */
+	@Nullable
+	private PreparedStatement WRITE_QUERY;
```

Comment:
```suggestion
	private @Nullable PreparedStatement WRITE_QUERY;
```
Nullable annotations can be placed directly before the type